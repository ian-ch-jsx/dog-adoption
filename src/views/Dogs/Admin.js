import { useState } from 'react';
import { addDog } from '../../services/dogs';
import DogForm from '../../components/Dog/DogForm';

export default function DogEdit() {
  const [name, setName] = useState('');
  const [species, setSpecies] = useState('');
  const [breed, setBreed] = useState('');
  const [age, setAge] = useState('');
  const [image, setImage] = useState('');
  const [bio, setBio] = useState('');

  const updateButton = async (e) => {
    e.preventDefault();
    await addDog({ name, species, breed, age, image, bio });
  };
  return (
    <div>
      <h1>Add pet:</h1>
      <DogForm
        name={name}
        setName={setName}
        species={species}
        setSpecies={setSpecies}
        breed={breed}
        setBreed={setBreed}
        age={age}
        setAge={setAge}
        image={image}
        setImage={setImage}
        bio={bio}
        setBio={setBio}
        updateButton={updateButton}
      />
    </div>
  );
}
