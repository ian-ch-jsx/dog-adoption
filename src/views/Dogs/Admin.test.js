import { render, screen } from '@testing-library/react';
import Admin from './Admin';

test('should render an edit form', async () => {
  const { container } = render(<Admin />);

  await screen.findByText('name:');

  expect(container).toMatchSnapshot();
});
