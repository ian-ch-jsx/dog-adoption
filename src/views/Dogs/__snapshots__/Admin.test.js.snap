// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an edit form 1`] = `
<div>
  <div>
    <h1>
      Add pet:
    </h1>
    <div
      class="update"
    >
      <form>
        <div
          class="control"
        >
          <label>
            name:
          </label>
          <input
            placeholder="Name"
            type="text"
            value=""
          />
        </div>
        <div
          class="control"
        >
          <label>
            species:
          </label>
          <input
            placeholder="species"
            type="text"
            value=""
          />
        </div>
        <div
          class="control"
        >
          <label>
            breed:
          </label>
          <input
            placeholder="Name"
            type="text"
            value=""
          />
        </div>
        <div
          class="control"
        >
          <label>
            age:
          </label>
          <input
            placeholder="age"
            type="number"
            value=""
          />
        </div>
        <div
          class="control"
        >
          <label>
            image(url):
          </label>
          <input
            placeholder="image"
            type="text"
            value=""
          />
        </div>
        <div
          class="control"
        >
          <label>
            bio:
          </label>
          <input
            placeholder="description"
            type="text"
            value=""
          />
        </div>
        <div
          class="button"
        >
          <button>
            Save
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
